### PR TITLE
Add some extra classes conditionally to the VideoPress modal.

### LINKED ISSUE #3636: Improve design of VideoPress view for Editor
Some good points made by Matt about the dialog for VideoPress introduced in https://github.com/Automattic/jetpack/pull/3556

videopress-view

- [x] put checkboxes to the left of the text, rather than floating all the way over to the right
- [x] we probably don't need a question mark on everything?
- [x] if something isn't editable, like the GUID, maybe don't put it in a form field. also "GUID" is pretty wonky
- [x] language on all of them can be made more intuitive or apparent

Other personal observations:
- [x] the word `flash` should be capitalized
- [x] the checkboxes should have the same style than other dialogs, like the link dialog
checkboxes

- [x] the embedded video should adjust to the width of the Editor. Right now is cropped as shown below.

responsive

### PR DESCRIPTION
**Pull request by georgestephanis**
Fixes #3636 

To each of the main `mce-container` field wrappers, this is adding either one or two classes.

The first is `mce-videopress-field-{$field_slug}` -- where slug is guid, w, at, hd, permalink, etc
The second is adding a `mce-videopress-checkbox` field to the fields that are checkboxes.

More coming from @jeffgolenski  as we consider overriding some styling and such -- this is just giving the selectors to base that work on top of.

### CODE REVIEW

**Comment by jeherve** [comment]
@jeffgolenski Let us know when this is ready for testing and review.

**Comment by jeffgolenski** [comment]
@jeherve Please take this for a spin. I made the adjustments as best I could without getting too sloppy (because of the Tiny MCE code). 

screen shot 2016-04-12 at 10 55 57 am

Also, I'm loading the styles as an inline-block (not many lines), so they only load where this modal is present. Let me know if I placed them in the right location within `shortcode.php`

cc @samhotchkiss @eliorivero @georgestephanis

**Comment by eliorivero** [comment]
@jeffgolenski I'm not 100% satisfied with this and it's my personal opinion 😄 the fields are too far from the labels, and when you're acting on a field, you have to move the hand towards the other side of the dialog to tick the checkboxes. I tried moving things around to follow core's layout and appearance and this is what I got:
dialog

The CSS to move the elements is very straightforward:
```css
.mce-videopress-checkbox .mce-checkbox {
    left: 120px !important;
    width: 100% !important;
}
.mce-videopress-checkbox .mce-label {
    left: 150px !important;
}
```
I narrowed the input fields to `width: 76px;` and the pixels and seconds are spans with
```css
.mce-videopress-checkbox .mce-label-unit {
    position: absolute;
    left: 210px;
    top: 5px;
}
```
 for the checkboxes is:
```css
.mce-videopress-checkbox i.mce-i-checkbox {
    border: 1px solid #b4b9be;
    background: #fff;
    color: #555;
    clear: none;
    cursor: pointer;
    display: inline-block;
    line-height: 0;
    height: 16px;
    margin: -4px 4px 0 0;
    outline: 0;
    padding: 0!important;
    text-align: center;
    vertical-align: middle;
    width: 16px;
    min-width: 16px;
    -webkit-appearance: none;
    -webkit-box-shadow: inset 0 1px 2px rgba(0,0,0,.1);
    box-shadow: inset 0 1px 2px rgba(0,0,0,.1);
    -webkit-transition: .05s border-color ease-in-out;
    transition: .05s border-color ease-in-out;
}

.mce-videopress-checkbox .mce-i-checkbox:before { {
    float: left;
    display: inline-block;
    vertical-align: middle;
    width: 16px;
    font: 400 21px/1 dashicons;
    speak: none;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    margin: -2px 0 0 -3px;
    content: "\f147";
    color: #1e8cbe;
}
```
Let me know what you think.

**Comment by jeffgolenski** [comment]
@eliorivero That looks great. Thanks for making that suggestion. I've added the necessary styles to make it happen, but 100% of the css you outlined above wasn't all necessary.

Just need a favor though. Can you commit the code changes to break apart the strings and inputs in the top, as you've outlined in your above screenshot?

screen shot 2016-04-12 at 7 10 06 pm

Just mention me after that and I'll do a final round of testing on some devices and we'll be good!

File: modules/videopress/shortcode.php [hunk 1]
```diff
@@ -195,16 +195,44 @@ function videopress_editor_view_js_templates() {
...
+			border: 1px solid #b4b9be;
+			background: #fff;
+			color: #555;
+			height: 16px;
+			width: 16px;
+			margin: -4px 4px 0 0;
+			vertical-align: middle;
+		}
+
+		.mce-videopress-checkbox .mce-i-checkbox:before { {
```
Comment (jeffgolenski) [inline modules/videopress/shortcode.php]: lines 224 – 235 I'm unsure at to what this is doing or why its needed. I committed it per your suggestion @eliorivero but just curious what it's doing seeing as though I'm not seeing the `.mce-i-checkbox:before` doing anything to the DOM

Comment (eliorivero) [inline modules/videopress/shortcode.php]: @jeffgolenski you're right, that's not needed, the correct selector is
`.mce-videopress-checkbox .mce-i-checkbox:before`

Comment (eliorivero) [inline modules/videopress/shortcode.php]: @jeffgolenski you're right, that's not needed, the correct selector is
`.mce-videopress-checkbox .mce-i-checkbox.mce-checked:before`

**Comment by eliorivero** [comment]
@jeffgolenski I went ahead and stylized the checkboxes and laid out the fields and their labels. Please have a look and amend whatever you consider necessary. I had to specify a fixed width for the dialog since it was collapsing due to the narrower fields. Would be good if @georgestephanis could quickly review how I added the `seconds` and `pixels` spans in `editor-view.js` using the `onopen`.
One thing I would add is to center the dialog since when I downsize the window, TinyMCE doesn't adjust its position. It's not critical so we can leave this for a future version.

**Comment by jeherve** [comment]
I played with this, and the improvements look nice. I think we could still improve a few other things, though:

### Flash

That old player doesn't include all the other options available in the modal, so it makes things a bit confusing. I shouldn't be able to select both "Open Source Codecs" and Flash, or Autoplay and Flash. If usage of the old flash player is not recommended, why adding it to the modal? Shouldn't we only leave it as a shortcode parameter instead, for legacy?

### Autoplay

Wouldn't it be less confusing to just call it "autoplay", or "autoplay this video", like YouTube and Vimeo do? "Autoplay video on page load seems more complicated than it should be.

screen shot 2016-04-13 at 2 50 08 pm

screen shot 2016-04-13 at 2 49 31 pm

I think "Loop" could also be simplified, like on Vimeo. Words like "page load" and "playback" will only make things more confusing for users imo.

### Layout issues

Although I use a theme including a post editor stylesheet (Twenty Sixteen), the VideoPress embed is larger than the editor. Is that to be expected?

screen shot 2016-04-13 at 2 42 57 pm

### Video offset

Would there be a way to make this more user-friendly?

screen shot 2016-04-13 at 2 45 36 pm

Right now, I can only enter seconds ("how many seconds is 1:35?"), and I can enter just about any random number.

**Comment by samhotchkiss** [comment]
@jeherve -- I like (and agree with) your feedback, however, I don't think any of them should be blockers to 4.0-- my vote would be to get this branch in as is, and then @georgestephanis and @dbtlr can update and improve in 4.1-- this is a better interface than exists now (obviously), and will only be seen by a very small number of users, so I think it is a move in the right direction, then we can continue to dial it in with future releases.

What say you?

**Comment by jeffgolenski** [comment]
I also agree with everything @jeherve said, and agree with @samhotchkiss as well. I'll create an issue with Jeremy's suggestions for the 4.1 release. 

Awesome work, @eliorivero – really appreciate the help!

**Comment by jeffgolenski** [comment]
Issue created: https://github.com/Automattic/jetpack/issues/3655

**Comment by dbtlr** [comment]
I agree with you @jeherve about the legacy flash player. I think we need to stop showcasing this as an option, as that player doesn't work in some browsers anymore. Hopefully we can start discussing any blockers for removing this ASAP.

**Comment by dereksmart** [comment]
I know it's not technically within the scope of this PR, but I think it's good practice to prevent scrolling to the background while a modal is open.

**Comment by dereksmart** [comment]
Something is broken with the `width` -- I set a video to 100px in the modal and it doesn't resize

**Comment by dereksmart** [comment]
I'm not seeing any videos rendered on the front end of my post -- should I be?

**Comment by eliorivero** [comment]
@dereksmart 
implemented your suggestion: prevent scrolling to the background while a modal is open.
fixed resizing of videos in editor

**Comment by jeffgolenski** [comment]
Working good for me in all current major browsers and older IE versions as well. :+1:

File: modules/videopress/shortcode.php [hunk 2]
```diff
@@ -158,8 +158,10 @@ function videopress_handle_editor_view_js() {
 		'modal_labels'      => array(
 			'title'     => __( 'VideoPress Shortcode', 'jetpack' ),
 			'guid'      => __( 'Video ID', 'jetpack' ),
-			'w'         => __( 'Video width (in pixels)', 'jetpack' ),
-			'at'        => __( 'Start video after (in seconds)', 'jetpack' ),
+			'w'         => __( 'Video Width', 'jetpack' ),
+			'w_unit'    => __( 'pixels', 'jetpack' ),
+			'at'        => __( 'Start Video After', 'jetpack' ),
```
Comment (georgestephanis) [inline modules/videopress/shortcode.php]: With `seconds` split off into a new string, this one should have more context for translators.

File: modules/videopress/js/editor-view.js [hunk 3]
```diff
@@ -209,8 +234,21 @@
 					}, renderer );
 
 					editor.insertContent( wp.shortcode.string( args ) );
+				},
+				onopen : function ( e ) {
```
Comment (georgestephanis) [inline modules/videopress/js/editor-view.js]: It may be better to move my overrides from modifying the FormItem.renderHtml prototype above into this.

Comment (georgestephanis) [inline modules/videopress/js/editor-view.js]: (not a blocker, we can totes do it later)

File: modules/videopress/shortcode.php (hunk 2 above)
Comment (eliorivero) [inline modules/videopress/shortcode.php]: Good one, added translation context. Thanks!

**Pull request merged**